src/controllers/legacycontrollersettings.cpp

```diff
@@ -221,7 +230,8 @@ LegacyControllerEnumSetting::LegacyControllerEnumSetting(
             !value.isNull();
             value = value.nextSiblingElement("value")) {
         QString val = value.text();
-        m_options.append(std::tuple<QString, QString>(val, value.attribute("label", val)));
+        QColor color = QColor(value.attribute("color"));
+        m_options.emplace_back(Item{val, value.attribute("label", val), color});
```

Comment:
`emplace_back` constructs the target type from the arguments in place, thus pass the args directly:
```suggestion
        QColor color = QColor(value.attribute("color"));
        m_options.emplace_back(val, value.attribute("label", val), color);
```
otherwise you'll construct a temporary and the move from it instead of a single ctor call.

Reply:
This doesn't work as there is no ctor on the struct. Do you want me to add one? Or replace with a `push_back`?

Reply:
[It does on compiler explorer](https://compiler-explorer.com/z/415acfGxe). I guess use push_back then, but I'd still be interested in the error message...

Reply:
What compiler version do you have?
![image](https://github.com/user-attachments/assets/712b5a57-df07-46b3-a38f-3a3d9d75204a)

![image](https://github.com/user-attachments/assets/2b5249ea-fba2-43ec-9fad-1d2e9ca2daa5)

Reply:
clang shipped with Ubuntu 22.04, which is 15 AFAIK. Is there any preprocessor I should put to support it going forward?

Reply:
Appreciate I could make it work, but I unfortunately also use my machine and toolchain for my full-time job, as well as other projects than Mixxx and I don't think I will take the risk to ~break~update it right now. Appreciate `main` runs on 24.04 as per the consensus, so completely okay with getting code to to compile and run stable there, but if I was to continue to contribute to Mixxx, I would need to get the code to compile on my setup too.

Are you happy to go with something like this?

```cpp
#if ((__GNUC__ && !__clang_major__ && __GNUC__ < 10) || (__clang_major__ && __clang_major__ < 16))
    myvec.emplace_back(Foo{1, 2});
#else
    myvec.emplace_back(1, 2);
#endif
```

Just want to put the emphasis on the fact that I'm not discussing the consensus on C++20 and appreciate other contributions will break Clang 14 eventually (that's the one on Ubuntu 22.04 btw).

Reply:
Oh lovely! Didn't know C++ features had macro directly available.

Reply:
yeah, always worth looking the original paper for a proposal ;)

Reply:
Looks like this might not work with MSVC... any suggestion?

Reply:
> C:\buildenv\mixxx-deps-2.5-x64-windows-release-c616147\installed\x64-windows-release\include\Qt6\QtCore\qarraydataops.h(570) : fatal error C1001: Internal compiler error.
Looks like a compiler error...  I'll have a look.

Reply:
Actually, MSVC on compiler explorer [doesn't support external libraries](https://github.com/compiler-explorer/compiler-explorer/issues/6681), so I don't really have an easy way to debug this. I guess we'll have to work around it. Lets just add `&& !defined(_MSC_VER)` so msvc uses the `push_back` path.